0035-ilp-over-http/0035-ilp-over-http.md

```diff
@@ -56,3 +52,115 @@ All ILP Packets MUST be returned with the HTTP status code `200: OK`.
 
 An endpoint MAY return standard HTTP errors, including but not limited to: a malformed or unauthenticated request, rate limiting, or an unresponsive upstream service. Connectors SHOULD either retry the request, if applicable, or relay an ILP Reject packet back to the original sender with an appropriate [Final or Temporary error code](./0027-interledger-protocol-4/0027-interledger-protocol-4#error-codes). Server errors (status codes 500-599) SHOULD be translated into ILP Reject packets with `T00: Temporary Error` codes.
 
+### Authentication
+When authenticating requests between Interledger nodes, it is important to choose an algorithm that maintains an appropriate balance between usability and security, while at the same time maintaining high-performance _and_ interoperability.
+
+In order to find this balance, this document defines two Authentication profiles, each with various trade-offs that should be considered before use:
+
+* `SIMPLE`: Allows two ILP nodes to utilize a previously agreed-upon shared-secret as a [Bearer token](https://tools.ietf.org/html/rfc6750) in all ILP-over-HTTP requests. Peers SHOULD consider this token to be opaque and SHOULD NOT derive any special meaning from the token. 
+
+* `JWT_HS_256`: Allows two ILP nodes to utilize a previously agreed-upon shared-secret in order to _derive_ a Bearer token that conforms to the JSON Web Token (JWT) specification. JWTs generated by this scheme can then be used as a Bearer in all ILP-over-HTTP requests.
+
+Peers MAY use any standard HTTP authentication mechanism to authenticate incoming requests, but SHOULD support both of the authentication profiles defined in this RFC, and MUST support `JWT_HS_256`.
+ 
+In addition, TLS Client Certificates are RECOMMENDED between peers for additional security. Basic authentication (username and password) is NOT RECOMMENDED for the same reasons that the `SIMPLE` profile is only reccommended for development and testing scenarios only.
```

Comment:
I'd consider TLS Client Certs to be a third authentication profile that I think negates the need for the others. It's the hardest to do but probably best because it doesn't require each request to be authenticated once the transport is authenticated.

Reply:
Hmmm...great point! I think I'll just move this to the end under "Recommendations" as something extra to consider for a secure infrastructure. 

I think that using TLS Client Certs _could_ negate the other profiles, but there are deployment scenarios where both are necessary (specifically, edge-terminated TLS connections, which are unfortunately _very_ common).

Checkout my new verbiage to see what you think.

Reply:
Hmmm...great point! I think I'll just move this to the end under "Recommendations" as something extra to consider for a secure infrastructure. 

I think that using TLS Client Certs _could_ negate the other profiles, but there are deployment scenarios where both are necessary (specifically, edge-terminated TLS connections, which are unfortunately _very_ common).

Checkout my new verbiage to see what you think.